pkg/api/vlabs/validate.go

```diff
@@ -25,7 +25,7 @@ func (o *OrchestratorProfile) Validate() error {
 	}
 
 	if o.OrchestratorType != Kubernetes &&
-		(o.KubernetesConfig.KubernetesImageBase != "" || o.KubernetesConfig.NetworkPolicy != "") {
+		(o.KubernetesConfig != nil) {
```

Comment:
The case of KubernetesConfig being present and KubernetesImageBase/NetworkPolicy  should also be handled.